cooked-validators/src/Cooked/MockChain/Monad/Direct.hs

```diff
@@ -395,324 +408,395 @@ runTransactionValidation slot parms utxoIndex signers txBodyContent consumedData
    in case mValidationError of
         Just err -> throwError (MCEValidationError err)
         Nothing -> do
-          -- Validation succeeded; now we update the indexes and the managed
-          -- datums. The new mcstIndex is just `newUtxoIndex`; the new
-          -- mcstDatums is computed by removing the datum hashes have been
-          -- consumed and adding those that have been created in the
-          -- transaction.
+          -- Validation succeeded; now we update the UTxO index, the managed
+          -- datums, and the managed Validators. The new mcstIndex is just
+          -- `newUtxoIndex`; the new mcstDatums is computed by removing the
+          -- datum hashes have been consumed and adding those that have been
+          -- created in the transaction.
           modify'
             ( \st ->
                 st
                   { mcstIndex = newUtxoIndex,
-                    mcstDatums = (mcstDatums st Map.\\ consumedData) `Map.union` producedData
+                    mcstDatums = (mcstDatums st Map.\\ consumedData) `Map.union` producedData,
+                    mcstValidators = mcstValidators st `Map.union` outputValidators
                   }
             )
 
           return (Pl.CardanoApiEmulatorEraTx cardanoTxModified)
 
-utxosSuchThat' ::
-  Monad m =>
-  (GenericOutput -> Bool) ->
-  MockChainT m [(Pl.TxOutRef, GenericOutput)]
-utxosSuchThat' outputPred = do
-  ix <- gets (Pl.getIndex . mcstIndex)
-  return . Map.toList
-    . Map.mapMaybeWithKey
-      ( \txOutRef output ->
-          let genericOutput = toGenericOutput output
-           in if outputPred genericOutput
-                then Just genericOutput
-                else Nothing
+ensureTxSkelOutsMinAda :: TxSkel -> TxSkel
```

Comment:
I presume the huge diff that follows is basically the result of uncommenting and contains no substantial changes, so I'm not looking too carefully.

BTW we might want to use multiline comments `{- -}` in the future to minimize diffs.